### PR TITLE
feat: rectify the external/internal recipients handling for RBF

### PR DESCRIPTION
**Pull request by rushilbg**
`verifyTransaction()` needed to be correctly filtering out the external recipients of the original tx to be replaced while verifying the built replacement tx.

BTC-820

### CODE REVIEW

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 1]
```diff
@@ -516,19 +516,19 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
...
     const allOutputs = [...explanation.outputs, ...explanation.changeOutputs];
 
     let expectedOutputs;
     if (txParams.rbfTxIds) {
       assert(txParams.rbfTxIds.length === 1);
-      const txToBeReplacedTransfer = await wallet.getTransfer({ id: txParams.rbfTxIds[0] });
+
+      const txToBeReplaced = await wallet.getTransaction({ txHash: txParams.rbfTxIds[0], includeRbf: true });
       // Note: Will work only when there is single transaction output per address
       // TODO: https://bitgoinc.atlassian.net/browse/BTC-826
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:527]:
```suggestion

```

**Review by OttoAllmendinger — APPROVED** [review]
nit

File: modules/abstract-utxo/src/abstractUtxoCoin.ts (hunk 1 above)
Comment (rushilbg) [reply modules/abstract-utxo/src/abstractUtxoCoin.ts:527]: Oh yeah using transaction doesn't warrant this TODO now.

Comment (rushilbg) [reply modules/abstract-utxo/src/abstractUtxoCoin.ts:527]: Raising PR.

**Pull request merged**